js/directives/lucca/user-picker.js

```diff
@@ -0,0 +1,322 @@
+				}
+				else {
+					scope.properties = DEFAULT_HOMONYMS_PROPERTIES;
+				}
+			}
+		};
+	})
+
+	.controller("luidUserPickerController", ['$scope', '$http', 'moment', '$timeout', '$q', function ($scope, $http, moment, $timeout, $q) {
+		var timeout = {};
+		var promise;
+		var maxUsers = 0;
+		var selectedUsersCount = 0;
+
+		$scope.user = {};
+		$scope.selected = {};
+		$scope.selected.users = [];
+
+		var getUsers = function(init, input) {
+			var formerEmployees = "formerEmployees=" + ($scope.showFormerEmployees ? "true" : "false");
+			var limit = "&limit=";
+			var clue = "&clue=" + input;
+			var query = "/api/v3/users/find?" + (init ? "" : (clue + "&")) + formerEmployees;
+
+			if (promise) {
+				promise.then(function (response) {}); // do nothing
+			}
+
+			// Cancel previous timeout
+			if (timeout.count) {
+				$timeout.cancel(timeout.count);
+			}
+
+			// If we use a UserPickerMultiple or a custom filter, we ask for all the users
+			if ((selectedUsersCount !== 0) || ($scope.useCustomFilter)) {
+				limit += MAGIC_NUMBER_maxUsers;
+			}
+			else {
+				limit += (MAX_COUNT + 1);
+			}
+			query += limit;
+
+			promise = $http.get(query);
+			promise.then(function (response) {
+				$scope.users = response.data.data.items;
+
+				// Handle pagination
+				if (!$scope.useCustomFilter && (selectedUsersCount === 0)) {
+					// If there are more results than MAX_COUNT, launch a timer to get the number of results
+					if ($scope.users.length > MAX_COUNT) {
+						countTimeout(input);
+						$scope.users = _.first($scope.users, MAX_COUNT);
+						$scope.users.push({ overflow: MAX_COUNT + "/..." });
+					}
+				}
+				else {
+					// Used for UserPickerMultiple only
+					if (selectedUsersCount > 0) {
+						// Remove duplicates between results and selected users
+						_.each($scope.selected.users, function (selectedUser) {
+							$scope.users = _.reject($scope.users, function (user) {
+								return (user.id === selectedUser.id);
+							});
+						});
+						// Number of results matching the query
+						$scope.count = $scope.users.length;
+					}
+
+					// Used when a custom filtering function is given
+					if ($scope.useCustomFilter) {
+						// Get new set of users
+						$scope.users = $scope.customFilter({ users: $scope.users });
+						$scope.count = $scope.users.length;
+					}
+
+					if ($scope.count > MAX_COUNT) {
+						$scope.users = _.first($scope.users, MAX_COUNT);
+						$scope.users.push({ overflow: MAX_COUNT + "/" + $scope.count });
+					}
+				}
+
+				if ($scope.showFormerEmployees) {
+					checkFormerEmployees();
+				}
+
+				if (hasHomonyms()) {
+					getHomonymsProperties();
+				}
+			});
+		};
+
+		var checkFormerEmployees = function () {
+			_.each($scope.users, function (user) {
+				if (moment(user.dtContractEnd).isBefore(moment())) {
+					user.isFormerEmployee = true;
+				}
+			});
+		};
+
+		var hasHomonyms = function () {
+			var hasHomonyms = false;
```

Comment:
i'm conflicted about if this method should flag the homonyms or just tell you there are some.

to flag the homonyms you do a double `_.each` so the complexity is `O(n^2)`. In a standard use case it's no problem cuz `$scope.users.length` is under 10 but with a custom filter and a big db `$scope.users.length` could be ~2k and this cn have serious performance impact.

if you use this method just to check if there are homonyms you can do it with a `_.uniq` which i think is in `O(n)` so the impact in perf is lessend.

especially when you consider that the presence of homonyms is the exception not the rule, the check to identify if we have a specific treatment to do 5% of the time must be as fast as possible

I think you should have something like that
```js
if(hasHomonyms()){
  handleHomonyms(getHomonyms());
}
```